README.md

```diff
@@ -1,58 +1,86 @@
 # Storage Extension Specification
 
 - **Title:** Storage
-- **Identifier:** <https://stac-extensions.github.io/storage/v1.0.0/schema.json>
+- **Identifier:** <https://stac-extensions.github.io/storage/v2.0.0/schema.json>
 - **Field Name Prefix:** storage
-- **Scope:** Item, Collection
+- **Scope:** Item, Catalog, Collection
 - **Extension [Maturity Classification](https://github.com/radiantearth/stac-spec/tree/master/extensions/README.md#extension-maturity):** Pilot
 - **Owner**: @davidraleigh @matthewhanson
 
 This document explains the Storage Extension to the [SpatioTemporal Asset Catalog](https://github.com/radiantearth/stac-spec) (STAC) specification.
-It allows adding details related to cloud storage access and costs to be associated with STAC Assets.
+It allows adding details related to cloud object storage access and costs to be associated with STAC Assets.
 This extension does not cover NFS solutions provided by PaaS cloud companies.
 
 - Examples:
-  - [Item example 1](examples/item-naip.json): Shows the basic usage of the extension in a STAC Item.
-  - [Item example 2](examples/item-nsl.json): Another example of basic usage.
+  - [NAIP Item](examples/item-naip.json): Shows the usage of the extension in combination with the alternate asset extension.
+  - [NSL Item](examples/item-nsl.json): Shows a mixture of storage providers, including custom S3 hosts.
+  - [Catalog with Link](examples/catalog-link.json): Shows the usage of the extension on a link in a STAC Catalog.
 - [JSON Schema](json-schema/schema.json)
 - [Changelog](./CHANGELOG.md)
 
 ## Fields
 
 The fields in the table below can be used in these parts of STAC documents:
 
+- [x] Catalogs
+- [x] Collections
+- [x] Item Properties (incl. Summaries in Collections)
+- [ ] Assets (for both Collections and Items, incl. Item Asset Definitions in Collections)
+- [ ] Links
+
+| Field Name        | Type                                                         | Description |
+| ----------------- | ------------------------------------------------------------ | ----------- |
+| `storage:schemes` | Map<string, [Storage Scheme Object](#storage-scheme-object)> | **REQUIRED.** A property that contains all of the storage schemes used by Assets and Links in the STAC Item, Catalog or Collection. |
+
+---
+
+The fields in the table below can be used in these parts of STAC documents:
+
 - [ ] Catalogs
 - [ ] Collections
-- [x] Item Properties (incl. Summaries in Collections)
+- [ ] Item Properties (incl. Summaries in Collections)
 - [x] Assets (for both Collections and Items, incl. Item Asset Definitions in Collections)
-- [ ] Links
+- [x] Links
+- [x] [Alternate Assets Object](https://github.com/stac-extensions/alternate-assets?tab=readme-ov-file#alternate-asset-object)
```

Comment:
Because the alternatives are checksum-compatible, i.e. usually have all the same metadata except for usually hreg, auth and storage. You' need to duplicate all metadata and could only identify the same assets based on the checksum, which many people don't provide.

Here's an example where it becomes more obious:

```json
    "B02": {
      "href": "https://example.com/asset_B02.jp2",
      "type": "image/jp2",
      "auth:refs": ["oidc"],
      "name": "HTTPS",
      "file:size": 134934911,
      "file:checksum": "16207bee94d1cf00e4427785eeca8cb0af8267c161d8229606d890d67297c0d033cb",
      "title": "Blue (band 2) - 10m",
      "eo:bands": [
        {
          "name": "B02",
          "common_name": "blue",
          "description": "Blue (band 2)",
          "center_wavelength": 0.492,
          "full_width_half_max": 0.267
        }
      ],
      "gsd": 10,
      "proj:epsg": 32612,
      "proj:shape": [
        10980,
        10980
      ],
      "proj:bbox": [
        399960.0,
        4990200.0,
        509760.0,
        5100000.0
      ],
      "proj:transform": [
        10.0,
        0.0,
        399960.0,
        0.0,
        -10.0,
        5100000.0
      ],
      "raster:bands": [
        {
          "nodata": 0,
          "data_type": "uint16",
          "bits_per_sample": 15,
          "spatial_resolution": 10,
          "scale": 0.0001,
          "offset": -0.1
        }
      ],
      "roles": [
        "data",
        "reflectance"
      ],
      "alternate": {
        "s3": {
          "href": "s3://example/asset_B02.jp2",
          "name": "S3",
          "auth:refs": ["s3"],
          "storage:refs": ["example"],
        } 
      },
    }
```

You wouldn't want to duplicate it all, right? :-)